src/openforms/forms/api/serializers/form_variable.py

```diff
@@ -21,23 +21,24 @@ def validate(self, attrs):
         static_data_keys = [item.key for item in FormVariable.get_static_data()]
 
         existing_form_key_combinations = []
-        for item in attrs:
+        errors = {}
+        for index, item in enumerate(attrs):
             key_form_combination = (item["key"], item["form"].slug)
             if key_form_combination in existing_form_key_combinations:
-                raise ValidationError(
-                    _("The form and key attributes must be unique together"),
-                    code="unique",
-                )
+                errors[f"{index}.key"] = _("Key not unique within form")
```

Comment:
Maybe use `serializers.ErrorDetail(_("The variable key must be unique within a form"), code="unique")` instead for more metadata?